R/validate_config.R

```diff
@@ -1,498 +1,152 @@
-has_required_cols <- function(cfg, df_name, required_colnames) {
```

Comment:
low priority for review as I added some components to validation, but much of the functionality is equivalent, just substantially reworked